src/plugins/index_pattern_management/public/components/edit_index_pattern/index_header/index_header.test.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import { createOpenSearchDashboardsReactContext } from '../../../../../opensearch_dashboards_react/public';
+import { coreMock, workspacesServiceMock } from '../../../../../../core/public/mocks';
+import { BehaviorSubject } from 'rxjs';
+import { WorkspaceObject } from 'opensearch-dashboards/public';
+import { IntlProvider } from 'react-intl';
+import { IndexHeader } from './index_header';
+
+describe('IndexHeader at new home page', () => {
+  const indexPattern = { id: 'default-index', title: 'Test Index Pattern', fields: [] };
+  const mockCoreStart = coreMock.createStart();
+  const workspaceObject = {
+    id: 'foo_id',
+    name: 'foo',
+  };
+  const getIndexHeader = (props: any) => {
+    const mockHeaderControl =
+      (props?.header as Function) ||
+      (() => {
+        return null;
+      });
+
+    const setDefault = jest.fn();
+    const refreshFields = jest.fn();
+    const deleteIndexPatternClick = jest.fn();
+    const { Provider } = createOpenSearchDashboardsReactContext({
+      ...mockCoreStart,
+      ...{
+        application: {
+          ...mockCoreStart.application,
+          capabilities: {
+            ...mockCoreStart.application.capabilities,
+            workspaces: { enabled: true },
+          },
+        },
+        uiSettings: { ...mockCoreStart.uiSettings, get: jest.fn().mockReturnValue(true) },
+        workspaces: {
+          ...workspacesServiceMock.createStartContract(),
+          currentWorkspace$: new BehaviorSubject<WorkspaceObject | null>(props?.workspace),
+        },
+        navigationUI: {
+          HeaderControl: mockHeaderControl,
+        },
+      },
+    });
+
+    return (
+      <IntlProvider locale="en">
+        <Provider>
+          <IndexHeader
+            setDefault={setDefault}
+            refreshFields={refreshFields}
+            deleteIndexPatternClick={deleteIndexPatternClick}
+            indexPattern={indexPattern}
+            defaultIndex={props?.defaultIndex}
+          />
+        </Provider>
+      </IntlProvider>
+    );
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders the set default button when index is not default and user is in workspace', () => {
```

Comment:
Shall we add a case that workspace is not enabled?

Reply:
Sure, I will add it